src/modules/navigator/navigator_main.cpp

```diff
@@ -742,8 +742,15 @@ void Navigator::geofence_breach_check(bool &have_geofence_position_data)
 			const float velocity_hor_abs = sqrtf(_local_pos.vx * _local_pos.vx + _local_pos.vy * _local_pos.vy);
 			_gf_breach_avoidance.setHorizontalVelocity(velocity_hor_abs);
 			_gf_breach_avoidance.setClimbRate(-_local_pos.vz);
-			test_point_distance = _gf_breach_avoidance.computeBrakingDistanceMultirotor();
-			vertical_test_point_distance = _gf_breach_avoidance.computeVerticalBrakingDistanceMultirotor();
+			// EDIT -- David Patrick 04/03/2022
+			// Hardcoded test_point distances to 0.2 to cause geofence breaches to be based on drone current position.
+			// These are effectively 0, however != 0 to prevent mathematical errors in bearing calculations elsewhere.
+			// Ensure that braking distance computations are still run for updating values.
+			test_point_distance = 0.2;
+			vertical_test_point_distance = 0.2;
+			_gf_breach_avoidance.computeBrakingDistanceMultirotor();
+			_gf_breach_avoidance.computeVerticalBrakingDistanceMultirotor();
+			// EDIT END
```

Comment:
Suggested alternative method as discussed on the phone - from line 778 do the following (or something like it - do check _global_pos_lat etc. are the right terms).
The advantage of this is that the change from default PX4 behaviour is much clearer to see, and is self-contained to a single part of the code, and hence will be easier to understand when we come back to this in a year, or try to recreate in future PX4 releases.
```
bool sees_geofence_behaviour = false;

if (sees_geofence_behaviour) { 
   // Sees preferred behaviour - just triggers based on drone current position
   // More predictable than default behaviour, but does result in geofence overshoot
	gf_violation_type.flags.dist_to_home_exceeded = !_geofence.isCloserThanMaxDistToHome(_global_pos.lat,
			_global_pos.long,
			_global_pos.alt);

	gf_violation_type.flags.max_altitude_exceeded = !_geofence.isBelowMaxAltitude(_global_pos.alt);

	gf_violation_type.flags.fence_violation = !_geofence.isInsidePolygonOrCircle(_global_pos.lat,
			_global_pos.long,
			_global_pos.alt);
} else { 
   // Default PX4 Behaviour - triggers using a look-ahead to predict where drone will be in the future 
   // More conservative (less likely to breach) but results in unpredictable behaviour as breach can retrigger as you try to recover the drone.

	gf_violation_type.flags.dist_to_home_exceeded = !_geofence.isCloserThanMaxDistToHome(fence_violation_test_point(0),
			fence_violation_test_point(1),
			_global_pos.alt);

	gf_violation_type.flags.max_altitude_exceeded = !_geofence.isBelowMaxAltitude(_global_pos.alt +
			vertical_test_point_distance);

	gf_violation_type.flags.fence_violation = !_geofence.isInsidePolygonOrCircle(fence_violation_test_point(0),
			fence_violation_test_point(1),
			_global_pos.alt);
}
```